src/App.js

```diff
@@ -1,16 +1,96 @@
 import React from "react";
+import axios from "axios";
 import logo from "./logo.png";
 import "./App.css";
 
 class App extends React.Component {
+  constructor(props) {
+    // Always call super with props in constructor to initialise parent class
+    super(props);
+    this.state = {
+      error: null,
+      value: "", // form input state
+      userInput: false,
+      city: "",
+      weather: "",
+      temperature: 0,
+      iconURL: "",
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    if (this.state.value === "") return; // skip the rest of code if user didn't input
```

Comment:
```suggestion
    if (!this.state.value) return; // skip the rest of code if user didn't input
```

empty strings are falsy